Fix pydantic-ai deprecation: mcp_servers -> toolsets

## Issue

The test suite is showing a deprecation warning from pydantic-ai:

```
tests/ai/test_mcp.py::test_mcp_git_server_tool_usage_and_output
  /home/runner/work/marvin/marvin/.venv/lib/python3.11/site-packages/pydantic_ai/agent/__init__.py:310: DeprecationWarning: `mcp_servers` is deprecated, use `toolsets` instead
    warnings.warn('`mcp_servers` is deprecated, use `toolsets` instead', DeprecationWarning)
```

## Context

Marvin has basic MCP (Model Context Protocol) client support to plug into MCP servers. Pydantic-ai has deprecated the `mcp_servers` parameter in favor of `toolsets`.

## Required Changes

Need to update all usages of `mcp_servers` to use the new `toolsets` API instead.

## Test Output
```
FAILED tests/ai/fns/test_extract.py::TestPydantic::test_extract_multiple_locations - Failed: Timeout (>60.0s) from pytest-timeout.
= 1 failed, 379 passed, 3 skipped, 8 xfailed, 2 xpassed, 6 warnings, 1 rerun in 170.42s (0:02:50)
```

--- a/src/marvin/agents/agent.py
+++ b/src/marvin/agents/agent.py
@@ -206,7 +206,7 @@ class Agent(Actor):
             agent_kwargs["tools"] = combined_tools
 
         if active_mcp_servers:
-            agent_kwargs["mcp_servers"] = active_mcp_servers
+            agent_kwargs["toolsets"] = active_mcp_servers
 
         agentlet = pydantic_ai.Agent[Any, Any](**agent_kwargs)
 

